controllers/httproute_controller.go

```diff
@@ -253,6 +253,30 @@ func (r *HTTPRouteReconciler) reconcileHTTPRouteResource(ctx context.Context, ht
 		r.eventRecorder.Event(httpRoute.K8sObject(), corev1.EventTypeWarning, k8s.HTTPRouteventReasonFailedAddFinalizer, fmt.Sprintf("Failed add finalizer due to %v", err))
 	}
 
+	backendRefIPFamiliesErr := r.getBackendRefsIpFamiliesError(ctx, httpRoute.Inner())
+
+	if backendRefIPFamiliesErr != nil {
+		httpRouteOld := httpRoute.Inner().DeepCopy()
+
+		setupRouteStatusParents(httpRoute.Inner())
```

Comment:
inverse ownership: mutating struct outside of struct, you should add a method to httpRoute with reference receiver:

```
func(r *httpRoute) setupRouteStatusParents() { ... }
```